File: binarySearchRecursive.py
def binarySearch(arr, l, r, x):
  """Interative"""
  while r >= l:
    mid = l + (r-l)//2
    if arr[mid] == x:
      return mid
    elif arr[mid] > x:
      r = mid - 1
    else:
      l = mid + 1
  return -1


#test
arr = [2,3,5,10,40]
x = 10

result = binarySearch(arr, 0, len(arr)-1, x)

if result != -1:
  print("Element is present at index %d"%result)
else:
  print("Element is not present in array")

File: binarySearchRecursive.java
class BinarySearch {
  int binarySearch(int arr[], int l, int r, int x) {
    if (r >= l) {
      int mid = l + (r - l)/2;
      if (arr[mid] == x) {
        return mid;
      }
      if (arr[mid] > x) {
        return binarySearch(arr, l, mid-1, x);
      }
      return binarySearch(arr, mid+1, r, x);  
    }
    return -1;
  }
}


class Main {
  public static void main(String[] args) {
    //System.out.println("Hello world!");
    BinarySearch ob = new BinarySearch();
    int arr[] = {2,3,5,10,50};
    int n = arr.length;
    int x = 10;
    int result = ob.binarySearch(arr, 0, n-1, x);
    if (result == -1) {
      System.out.println("Element not present");
    }
    else {
      System.out.println("Element found at index " + result);
    }
  }
}
